src/software/logger/protobuf_sink.cpp

```diff
@@ -0,0 +1,72 @@
+#include "software/logger/protobuf_sink.h"
+
+#include <chrono>
+
+#include "base64.h"
+#include "google/protobuf/text_format.h"
+#include "proto/robot_log_msg.pb.h"
+
+ProtobufSink::ProtobufSink()
+{
+    // Setup the logs
+    unix_senders["log"] = std::make_unique<ThreadedUnixSender>(UNIX_BASE_PATH + "log");
+}
+
+void ProtobufSink::sendProtobuf(g3::LogMessageMover log_entry)
+{
+    auto level = log_entry.get()._level;
+
+    if (level.value == VISUALIZE.value)
+    {
+        std::string msg = log_entry.get().message();
+        size_t pos      = msg.find(TYPE_DELIMITER);
+
+        std::string proto_type_name  = msg.substr(0, pos);
+        std::string serialized_proto = msg.substr(pos + TYPE_DELIMITER.length());
+
+        // If we don't already have a unix sender for this type, lets create it
+        if (unix_senders.count(proto_type_name) == 0)
+        {
+            unix_senders[proto_type_name] =
+                std::make_unique<ThreadedUnixSender>(UNIX_BASE_PATH + proto_type_name);
+        }
+
+        // Send the protobuf
+        unix_senders[proto_type_name]->sendString(serialized_proto);
+    }
+    else
+    {
+        TbotsProto::RobotLog log_msg_proto;
+        TbotsProto::LogLevel log_level_proto;
+
+        if (TbotsProto::LogLevel_Parse(log_entry.get().level(), &log_level_proto))
+        {
+            std::time_t t = std::time(0);
+            log_msg_proto.mutable_created_timestamp()->set_epoch_timestamp_seconds(
+                static_cast<double>(t));
```

Comment:
why don't we use `std::chrono::time_point<std::chrono::steady_clock>` here?

Reply:
TIL about time_point, changed